### DISCUSSION TITLE
Routerfleet Platform

### DISCUSSION THREAD

**Discussion by abbio90** [eduardogsilva/routerfleet#51] [Ideas]
Hello, I recently discovered routerfleet. Very interesting, but I noticed that some Mikrotik branded devices cannot be used because your tool checks the platform which must be strictly "Mikrotik". Is it possible to modify the code to allow Mikrotik branded routers? Maybe where the platform is ignored and the user declares that it is a Mikrotik device?

**Comment by matteocisilino** [comment]
Hello Abbio90 , 
as a newbee i'm exploring the project and data generated, as i can say and I suppose Routerfleet does is a simple ssh shell that has two steps : 
1. launch a "/export show-sensitive file=$FILE" where file is a standard definition 
2. wait a time T to be sure about file creation 
3. launch an scp to move local the generated script config file
4. Routefleet does its magic :)

so after these lines , if the router is a reBranded based on RouterOS or SwitchOS ( not the lite version ) can run terminal commands. If the rebranded is "closed" like ( here in italy ) Vodafone PowerStations , you don't have direct access to terminal , but only using RESTAPI via web JSON :'(

**Comment by eduardogsilva** [comment]
Hello there!

Apologies for the delayed response—I've been swamped over the past few weeks.

When I first got the notification, I did a quick search about rebranded MikroTik routers but couldn’t find any solid information. Unfortunately, I forgot to come back here..

@matteocisilino, you’ve got the right idea! Let me just add a couple of points to clarify:

1- The export command can also be configured in the backup profile to include sensitive and/or terse data.
-2 The pause for retrieval can also be set within the profile. If I recall correctly, the default is 60 seconds.

As for RouterFleet, it uses SSH+SCP for executing and retrieving backups. If the rebranded router doesn’t support SSH, unfortunately, it won’t be compatible.

I’m planning to add compatibility with other brands, but testing and maintaining feature parity without purchasing a ton of routers is a challenge I still need to address.

Cheers!
